src/DocumentType/AbstractDocument.php

```diff
@@ -17,23 +20,48 @@ abstract class AbstractDocument implements DocumentInterface
 {
     public function getDocumentType(): ?string
     {
-        if ($this->getType() !== DocumentInterface::TYPE_DOCUMENT) {
+        if (!in_array($this->getType(), [DocumentInterface::TYPE_DOCUMENT, DocumentInterface::TYPE_ASSET], true)) {
             return null;
         }
 
-        $candidate = [
-            PimcoreDocument\Folder::class => 'folder',
-            PimcoreDocument\Page::class => 'page',
-            PimcoreDocument\Snippet::class => 'snippet',
-            PimcoreDocument\Link::class => 'link',
-            PimcoreDocument\Hardlink::class => 'hardlink',
-            PimcoreDocument\Email::class => 'email',
-            PimcoreDocument\Newsletter::class => 'newsletter',
-            PimcoreDocument\Printpage::class => 'printpage',
-            PimcoreDocument\Printcontainer::class => 'printcontainer',
-        ][$this->getSubType()] ?? null;
-
-        if ($candidate === null || !in_array($candidate, PimcoreDocument::getTypes(), true)) {
+        $candidate = null;
+
+        if ($this->getType() === DocumentInterface::TYPE_DOCUMENT) {
+            $candidate = [
+                PimcoreDocument\Folder::class => 'folder',
+                PimcoreDocument\Page::class => 'page',
+                PimcoreDocument\Snippet::class => 'snippet',
+                PimcoreDocument\Link::class => 'link',
+                PimcoreDocument\Hardlink::class => 'hardlink',
+                PimcoreDocument\Email::class => 'email',
+                PimcoreDocument\Newsletter::class => 'newsletter',
+                PimcoreDocument\Printpage::class => 'printpage',
+                PimcoreDocument\Printcontainer::class => 'printcontainer',
+            ][$this->getSubType()] ?? null;
+
+            if (!in_array($candidate, PimcoreDocument::getTypes(), true)) {
+                throw new UnknownPimcoreElementType($candidate);
+            }
+        }
+
+        if ($this->getType() === DocumentInterface::TYPE_ASSET) {
+            $candidate = [
+                PimcoreAsset\Archive::class => 'archive',
+                PimcoreAsset\Audio::class => 'audio',
+                PimcoreAsset\Document::class => 'document',
+                PimcoreAsset\Folder::class => 'folder',
+                PimcoreAsset\Image::class => 'image',
+                PimcoreAsset\Text::class => 'text',
+                PimcoreAsset\Unknown::class => 'unknown',
+                PimcoreAsset\Video::class => 'video',
+            ][$this->getSubType()] ?? null;
```

Comment:
@limenet I have to implement a list that searches acrosss all of these types. Previously I returned the `Asset` class itself from the `getSubType` method, but now it is not working anymore, because it is not listed in this list. Can we just add another line in this array like this:

```php
[
    PimcoreAsset::class => 'assets',
    // other entries
]
```

Reply:
An alternative would be to allow the `getSubType` method to return `null`. This would cause the return type to change and probably needs some special handling in the `AbstractDocument` class.

Reply:
Assets and documents are similarly handled by Pimcore itself and since we don't support that use case for documents in this package, I'd rather not add it for assets.

> An alternative would be to allow the `getSubType` method to return `null`.

This is something we could consider for v2 but not for a backport since this might introduce breaking changes and v2 being just around the corner, it's a considerable effort to backport it (even implementing it for v2 isn't as simple as just changing the method signature).

Since the sub types are a fairly small set of values, have you considered creating a `IndexDocument` for every asset type (`Video`, `Archive` etc.)?

Reply:
see #42